autosklearn/metalearning/metafeatures/metafeatures.py

```diff
@@ -1002,7 +1002,9 @@ def calculate_all_metafeatures(X, y, categorical, dataset_name, logger,
                 # sparse matrices because of wrong sparse format)
                 sparse = scipy.sparse.issparse(X)
                 DPP = DataPreprocessor(
-                    categorical_features=categorical, force_sparse_output=True)
+                    feat_type={i: 'categorical' if feat else 'numerical'
```

Comment:
Could you please add type annotations to the data preprocessor and all the other changes you do in this PR? This PR is a bit hard to review and I believe that now would be a good time to add type annotations to improve readability.

Reply:
I have added mypy support to preprocessing!